backend/de.metas.manufacturing/src/main/java/de/metas/manufacturing/generatedcomponents/IComponentGenerator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * #%L
+ * de.metas.manufacturing
+ * %%
+ * Copyright (C) 2020 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.manufacturing.generatedcomponents;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import lombok.NonNull;
+import org.adempiere.mm.attributes.AttributeCode;
+import org.adempiere.mm.attributes.api.ImmutableAttributeSet;
+
+public interface IComponentGenerator
+{
+
+	ImmutableAttributeSet generate(int qty, @NonNull ComponentGeneratorParam parameters, @NonNull ImmutableAttributeSet existingAttributes);
```

Comment:
maybe it's by OCD :) but i would suggest to have the `ComponentGeneratorParam parameters` param first on last because basically there u have 2 types of params: the ones which are coming from the caller (qty, existingAttributes) and the one which is coming from API (parameters)

Reply:
makes sense.

Done